### PR TITLE
Bump minimal Terraform version 1.0 -> 1.2

### PR DESCRIPTION
**Pull request by mr0re1**
### Submission Checklist

* [ ] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [ ] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [ ] Is unit test coverage still above 80%?
* [ ] Have you updated all applicable documentation?
* [ ] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Review by tpdownes — DISMISSED** [review]
I understand that we are generally bumping our recommended Terraform to 1.2.

I believe the gke-node-pool module requires 1.2+ for interpolation within the error_message. That said, I do not see why the gke-cluster module is being bumped. If there's a good reason, please merge. If there is not, please remove the change and then merge. 
At the same time, I don't see any features in the two GKE modules that require the upgrade. Am I not seeing something?

**Comment by tpdownes** [comment]
> I understand that we are generally bumping our recommended Terraform to 1.2.
> 
> I believe the gke-node-pool module requires 1.2+ for interpolation within the error_message. That said, I do not see why the gke-cluster module is being bumped. If there's a good reason, please merge. If there is not, please remove the change and then merge. At the same time, I don't see any features in the two GKE modules that require the upgrade. Am I not seeing something?

I may have reversed the module names in the above comment. My point is that I only see the justification for going to 1.2 for one of these modules.

**Comment by tpdownes** [comment]
A by-eye search suggests the following modules may also need 1.2+

- vm-instance
- batch-login-node
- batch-job-template
- dashboard

**Review by ankit-gautam23 — APPROVED** [review]
LGTM

**Comment by mr0re1** [comment]
Reverted changes on modules, making this PR into "update ghpc-wide required terraform version".
Module-specifc changes can be resolved independently.

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # PREAMBLE
```
Comment (cboneti) [inline Makefile:1]: Nit: don't you want to already move Packer to 1.8 here?

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 MIN_PACKER_VERSION=1.6 # for building images
-MIN_TERRAFORM_VERSION=1.0 # for deploying modules
+MIN_TERRAFORM_VERSION=1.2 # for deploying modules
```
Comment (cboneti) [inline Makefile:3]: This is ok, but you should probably update https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/pkg/modulewriter/tfversions.go#L19 as well.

**Review by cboneti — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (mr0re1) [reply Makefile:1]: I would bump it to `1.7.9` instead to match requirements in `modules/packer/custom-image/versions.pkr.hcl`, but I would do it separately and after better version comparator is deliverd.

**Review by cboneti — APPROVED** [review]

**Pull request merged**